### PR TITLE
7414: Authorize policies fix

### PR DESCRIPTION
**Pull request by RuthBerry96**
Story: [AB#7103](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/7103)
Task: [AB#7414](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/7414)

### CODE REVIEW

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Update Description.feature [hunk 1]
```diff
@@ -82,15 +82,23 @@ Scenario: 8. A buyer user cannot update an orders description for an organisatio
     Then a response with status code 403 is returned
 
 @5322
-Scenario: 9. Service Failure
+Scenario: 9. A user with read only permissions, cannot updated an orders description
```
Comment (sam-roberts-bjss) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Update Description.feature:85]: update rather than updated?

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Ordering Party Update .feature [hunk 2]
```diff
@@ -76,30 +76,72 @@ Scenario: 2. Updating an ordering party with boundary values results in persiste
...
-Scenario: 3. A non existent orderId returns not found
-    When the user makes a request to retrieve the ordering-party section with the ID INVALID
-    Then a response with status code 404 is returned
+Scenario: 3. Updating an ordering party, with a non existent model returns not found
+    When the user makes a request to update the order party with order ID C000014-01 with no model
+    Then a response with status code 400 is returned
 
 @4616
-Scenario: 4. If a user is not authorised then they cannot access the ordering-party
+Scenario: 4. If a user is not authorised then they cannot update the ordering-party
```
Comment (sam-roberts-bjss) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Ordering Party Update .feature:84]: Wondered whether it would be possible to simplify this test as the user is not logged in can we just send a simple request that doesn't contain all the information of an order record?

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Supplier Section Update.feature [hunk 3]
```diff
@@ -116,7 +116,21 @@ Scenario: 6. A buyer user cannot update a supplier section for an organisation t
     Then a response with status code 403 is returned
 
 @4621
-Scenario: 7. Service Failure
+Scenario: 7. A user with read only permissions cannot update a supplier section
+    Given the user is logged in with the Readonly-Buyer role for organisation e6ea864e-ef1b-41aa-a4d5-04fc6fce0933
```
Comment (sam-roberts-bjss) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Supplier Section Update.feature:120]: Similar thing as the previous comment. Could this test be simplified to not include most of the order information?

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Update Description.feature (hunk 1 above)
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Order Update Description.feature:85]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Supplier Section Update.feature (hunk 3 above)
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Supplier Section Update.feature:102]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Ordering Party Update .feature (hunk 2 above)
Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/Ordering Party Update .feature:84]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/SupplierSectionSteps.cs [hunk 4]
```diff
@@ -104,15 +104,16 @@ public async Task ThenTheResponseContainsTheFollowingSupplierAddress(Table table
         public async Task WhenTheUserMakesARequestToUpdateTheSupplierWithOrderId(string orderId, Table table)
         {
             var supplierTable = table.CreateInstance<SupplierSectionTable>();
-            var addressTable = _context["SupplierAddress"];
-            var contactTable = _context["SupplierContact"];
+            
+            _context.TryGetValue("SupplierAddress", out var address);
```
Comment (sam-roberts-bjss) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/SupplierSectionSteps.cs:108]: For consistency reasons want to ensure the code is using the same key e.g. "SupplierAddress" to retrieve a value as well as setting a value. Could `SupplierAddress` & `SupplierContact` be changed to use constants? It should be the same constant wherever the data is added too.

Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/SupplierSectionSteps.cs]: 👍

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/SupplierSectionSteps.cs [hunk 5]
```diff
@@ -160,6 +162,20 @@ public async Task ThenTheSupplierForOrderIsUpdated(string orderId, Table table)
             actual.Should().BeEquivalentTo(supplier);
         }
 
+        [Given(@"the user wants to update the supplier address section")]
+        public void WhenTheUserWantsToUpdateTheSupplierAddressSection(Table table)
+        {
+            var address = table.CreateSet<SupplierAddressTable>().FirstOrDefault();
```
Comment (sam-roberts-bjss) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/SupplierSectionSteps.cs:168]: This could use `table.CreateInstance<SupplierAddressTable>()` rather than `CreateSet`. This would mean we don't need the `FirstOrDefault` call.

Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Steps/SupplierSectionSteps.cs:168]: 👍
